### PR TITLE
measure API execute time seen from guestlib

### PR DESCRIPTION
**Pull request by photoszzt**
Signed-off-by: Zhiting Zhu <zhitingz@cs.utexas.edu>

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Document update (this change is mainly a documentation update)

## Checklist:

- [x] My code passes format and lint checks.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] My changes generate no new warnings.
- [x] I have tested my code with a reasonable workload.
- [ ] My code **may break** some other features.

### CODE REVIEW

**Comment by photoszzt** [comment]
The parse script outputs data like this:
image

File: generate.py [hunk 1]
```diff
@@ -39,27 +39,29 @@ def download_llvm_lib():
...
-        logger.info(f"ava-llvm already unpacked to {CLANG_LIB_DIR}")
+        logger.info("ava-llvm already unpacked to %s", CLANG_LIB_DIR)
 
 
 CAVA_DIR = os.path.dirname(os.path.realpath(sys.argv[0])) + "/cava"
-CUDA_10_1_CFLAGS = "-I/usr/local/cuda-10.1/include".split(" ")
+CUDA_10_1_CFLAGS = ["-I/usr/local/cuda-10.1/include"]
 GLIB2_CFLAGS = pkgconfig.cflags("glib-2.0").split(" ")
 FMT_CFLAGS = ["-I" + os.path.dirname(os.path.realpath(sys.argv[0])) + "/third_party/fmt/include"]
+GSL_CFLAGS = ["-I" + os.path.dirname(os.path.realpath(sys.argv[0])) + "/third_party/GSL/include/"]
```
Comment (yuhc) [inline generate.py:63]: `+ "/third_party/GSL/include"]`

File: generate.py [hunk 2]
```diff
@@ -39,27 +39,29 @@ def download_llvm_lib():
     if download_llvm:
         wget.download(LLVM_URL, download_target)
         print()
-        logger.info(f"ava-llvm downloaded to {download_target}")
```
Comment (yuhc) [inline generate.py:42]: Keep f strings.

File: generate.py [hunk 3]
```diff
@@ -49,17 +49,19 @@ def download_llvm_lib():
     if download_llvm or unzip_llvm:
         with tarfile.open(download_target) as f:
             f.extractall(LLVM_DIR)
-        logger.info(f"ava-llvm unpacked to {CLANG_LIB_DIR}")
+        logger.info("ava-llvm unpacked to {CLANG_LIB_DIR}")
```
Comment (yuhc) [inline generate.py:52]: Revert this line.

File: guestlib/queue_worker.h [hunk 4]
```diff
@@ -0,0 +1,21 @@
+#ifndef _AVA_GUESTLIB_QUEUE_WORKER_H_
+#define _AVA_GUESTLIB_QUEUE_WORKER_H_
+
+#include "common/cmd_channel.hpp"
+#include "common/declaration.h"
+#include "common/linkage.h"
+
+namespace ava {
+
+class EXPORTED QueueWorker {
```
Comment (yuhc) [inline guestlib/queue_worker.h:10]: Should it be `CmdQueue`  and define in `common/cmd_queue.h`?
(API server may use the same interface to implement reply command queue in the future.)

File: guestlib/extensions/command_batch_worker.h [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+#include <glib.h>
+
+#include "common/cmd_channel.hpp"
+#include "common/support/thread.h"
+#include "guestlib/guest_thread.h"
+#include "guestlib/queue_worker.h"
+
+namespace ava {
+
+class CmdBatchingWorker : public QueueWorker {
```
Comment (yuhc) [inline guestlib/extensions/command_batch_worker.h:13]: `Worker` is a confusing naming, as the API server is also called "worker". Maybe `GeustCmdBatchingQueue: public CmdQueue`?

**Review by yuhc — CHANGES_REQUESTED** [review]

**Review by yuhc — APPROVED** [review]

**Pull request merged**